Content.Server/_NF/PacifiedZone/PacifiedlZoneGeneratorSystem.cs

```diff
@@ -0,0 +1,94 @@
+using Robust.Server.GameObjects;
+using Robust.Shared.Timing;
+using Content.Shared.Humanoid;
+using Content.Shared.CombatMode.Pacification;
+using Content.Shared.Mind;
+using Content.Shared.Roles;
+using Content.Shared.Roles.Jobs;
+
+
+namespace Content.Server._NF.PacifiedZone
+{
+    public sealed class PacifiedZoneGeneratorSystem : EntitySystem
+    {
+        [Dependency] private readonly UserInterfaceSystem _userInterface = default!;
+        [Dependency] private readonly EntityLookupSystem _lookup = default!;
+        [Dependency] private readonly IGameTiming _gameTiming = default!;
+        [Dependency] private readonly IEntityManager _entMan = default!;
+        [Dependency] private readonly SharedMindSystem _mindSystem = default!;
+        [Dependency] private readonly SharedJobSystem _jobSystem = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<PacifiedZoneGeneratorComponent, ComponentInit>(OnComponentInit);
```

Comment:
SubscribeLocalEvent<PacifiedByZoneComponent, ComponentStartup>(OnStartup);
       SubscribeLocalEvent<PacifiedByZoneComponent, ComponentShutdown>(OnShutdown);
For example, you can add two tracking the appearance of a new component here.